api/src/pcapi/alembic/versions/20230302T174724_bc9452222a80_add_not_null_constraint_on_venue_venueTypeCode_step_4_of_4.py

```diff
@@ -0,0 +1,20 @@
+"""Add NOT NULL constraint on "venue.venueTypeCode" (step 4 of 4)"""
+from alembic import op
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "bc9452222a80"
+down_revision = "d66070f570d2"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_constraint("venue_venueTypeCode_not_null_constraint", table_name="venue")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Pourquoi in drop la contrainte à la fin ?

J'imagine que c'est parce qu'elle est overridée par la "vrai" au niveau de la table ?

Reply:
On supprime la contrainte au niveau de la _table_, car l'étape 3 créé une contrainte au niveau de la _colonne_ (dont un des avantages est qu'elle apparaît plus clairement quand on fait `\d table` dans un client PostgreSQL). Pour plus d'infos sur ces 4 étapes, tu peux jeter un oeil aux commentaires dans le script de génération de ces 4 migrations : https://github.com/pass-culture/pass-culture-main/blob/b2fddf0eb75ff17f74dd45670c4d1e45e7775b12/api/bin/alembic_add_not_null_constraint.py#L48-L113